src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -1475,11 +1573,17 @@ void MacroAssembler::kernel_crc32(Register crc, Register buf, Register len,
 
   const ExternalAddress table_addr = StubRoutines::crc_table_addr();
   la(table0, table_addr);
-  add(table1, table0, 1*256*sizeof(juint), tmp1);
-  add(table2, table0, 2*256*sizeof(juint), tmp1);
-  add(table3, table2, 1*256*sizeof(juint), tmp1);
+  add(table1, table0, 1*single_talbe_size*sizeof(juint), tmp1);
+  add(table2, table0, 2*single_talbe_size*sizeof(juint), tmp1);
+  add(table3, table2, 1*single_talbe_size*sizeof(juint), tmp1);
 
+  if (UseRVV) {
+    const int64_t tmp_limit = MaxVectorSize >= 32 ? unroll_words*2 : unroll_words*4;
+    sub(tmp1, len, tmp_limit);
+    bge(tmp1, zr, L_vector_entry);
+  }
```

Comment:
Hi Hamlin, I think maybe we should introduce another assember routine for the vector code? Let's say `kernel_crc32_using_vector` and delegate the work to it under `UseRVV`. That seems more cleaner to me and avoids "offset is too large" issue. I will take a look at the vector code. BTW: Should `single_talbe_size` be `single_table_size`?

Reply:
Not sure if I understand your suggestion correctly. Do you mean something like below?
```
address generate_updateBytesCRC32() {
  if (UseRVV) { kernel_crc32_using_vector(); }
  else { kernel_crc32(...); }
}
```
But as kernel_crc32_using_vector reuses the code in kernel_crc32, and even with UseRVV, in some condition (when size is not large enough) we still need to fallback to L_unroll_loop_entry.
Or maybe I could misunderstand what you mean?

Reply:
In a summary, the code paths are went through in following order: vector(optional) -> loop unroll -> other scalar cases, it depens on data size + UseRVV. So in UseRVV case, we need all the code path.

Reply:
Ah, I see. Regarding the "offset is too large" issue, could the far versions of these branches help? I mean setting the `is_far` parameters to true [1].

[1] https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/riscv/macroAssembler_riscv.hpp#L670-L675

Reply:
I was thinking the similar thing.
But it's bit tedious, as I need to figure out which jump should be far and which not, or I could just simply change any where suspicious, but it's not good either.
So I pick the simplest solution to move it to the end, in other intrinsics I saw the similar code layout, they don't have the comment for it, but I guess they are doing the similar things.

Reply:
I am just a bit concerned about the two `j(L_exit);` added. Seems the one in the `if (UseRVV)` block is not necessary?

Reply:
Thanks, fixed.